aten/src/ATen/templates/OpsAlreadyMovedToC10.cpp

```diff
@@ -1,25 +1,46 @@
 #include <ATen/core/OpsAlreadyMovedToC10.h>
 
+#include <string>
+#include <cstring>
+#include <utility>
+#include <unordered_set>
+#include <ATen/core/operator_name.h>
+
 // ${generated_comment}
 
+// TODO Once all ATen ops are moved to c10, this file should be removed
+
 namespace at {
 
-// TODO Once all ATen ops are moved to c10, this file should be removed
+namespace {
+struct OpNameEquals final {
+  bool operator()(const std::pair<const char*, const char*>& lhs, const std::pair<const char*, const char*>& rhs) const {
+      return 0 == strcmp(lhs.first, rhs.first) && 0 == strcmp(lhs.second, rhs.second);
+  }
+};
+
+struct OpNameHash final {
+  size_t operator()(const std::pair<const char*, const char*>& p) const {
+      // use std::hash<std::string> because std::hash<const char*> would hash pointers and not pointed-to strings
+      return std::hash<std::string>()(p.first) ^ (~ std::hash<std::string>()(p.second));
```

Comment:
I understand that we can't use the `const char*` hasher here, but doesn't this mean that each attempt to hash will actually have to allocate `std::string`s for both values, and copy their contents, only to throw them away once they're hashed? This seems quite expensive. We already have `hash_combine`, so maybe it's better to use this instead?